angular/src/directives/navigation/ion-router-outlet.ts

```diff
@@ -308,10 +308,10 @@ export class IonRouterOutlet implements OnDestroy, OnInit {
     }
 
     this.activatedView = enteringView;
-    this.stackCtrl.setActive(enteringView).then((data) => {
-      this.navCtrl.setTopOutlet(this);
+
+    this.navCtrl.setTopOutlet(this, this.stackCtrl.setActive(enteringView)).subscribe((stackEvent) => {
```

Comment:
~ An alternative pattern here that may be more clear, is having an API like:
```ts
this.navCtrl.setTopOutlet(this, enteringView)
```

Then providing the stack controller to the nav controller, so that it can call the `setActive` method. This would cut down on passing around a promise, which is an uncommon pattern. 

I will explore this more.

Reply:
Opted against this. The `stackCtrl` isn't an Angular provider, it's a constructed class local to each router outlet. Trying to pass the instance through to the `setTopOutlet` implementation led to failing tests and inconsistent results.